src/Approvals/SFA.DAS.Approvals.Api/Controllers/CohortController.cs

```diff
@@ -252,4 +253,38 @@ public async Task<IActionResult> SelectLegalEntity(long accountId)
             return BadRequest();
         }
     }
+
+    [HttpGet]
+    [Route("provider/{providerId}/unapproved/add/select-employer")]
+    public async Task<IActionResult> GetSelectEmployer(
+        [FromRoute] int providerId,
+        [FromQuery] string searchTerm,
+        [FromQuery] string sortField,
+        [FromQuery] bool reverseSort,
+        [FromQuery] bool useLearnerData)
+    {
+        try
+        {
+            var result = await mediator.Send(new GetSelectEmployerQuery
+            {
+                ProviderId = providerId,
+                SearchTerm = searchTerm,
+                SortField = sortField,
+                ReverseSort = reverseSort,
+                UseLearnerData = useLearnerData
+            });
+
+            if (result == null)
+            {
+                return NotFound();
+            }
+
+            return Ok((GetSelectEmployerResponse)result);
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, "Error in GetSelectEmployer for provider {ProviderId}", providerId);
+            return BadRequest();
+        }
+    }
```

Comment:
Tests around this are failing and no test for exception code (but this is not a big issue)

Reply:
fixed